### DISCUSSION TITLE
not refresh ISR  files into kubernetes

### DISCUSSION THREAD

**Discussion by fx69005** [vercel/next.js#17886] [Help]
Hello , we have a problem with ISR into our k8 . statics files are not refresh every 500 ms . but when i restart the container  static files are refreshed 
has anyone encountered this problem? 
next.js v9.5.4

**Comment by valse** [comment]
I have a similar problem but with a Windows server node process: the page on the preview mode works fine but, after publishing it, the public page response with the 404 UI because the ISR never revalidate the page... I set a revalidate time of 60 seconds; it seems that set a lower value like 1 second it works but my backend API still under requests.

Only a restart of the process or deleting the pages created under the `server/pages` folder, let the regeneration of the missing published pages.

I'm using latest nextjs version with the `unstable_blocking` fallback

**Comment by aamorozov** [comment]
This might be related to auto-scaling(if you have it enabled). Once a user hits a page that has its data changed, that page will be revalidated, but only for the app running in one pod. Other pods might have the stale content until another user/same user with a different session hits that page again and it will be served from one of those other pods. The solution for this would be to either use a shared cache of Next build across the pods or force a rebuild of the app upon the data change(less preferred usually, due to the increased time from content change to users seeing the updated content) but easier to set up

**Reply by eglaost** [reply]
@aamorozov with cache are you referring to the cache directory inside the build directory?

**Reply by aamorozov** [reply]
You'll need to have the whole build output shared as that would have the BUILD_ID and all assets that are used in revalidation/static assets. So that you can sync it between the pods/servers depending on how you perform scaling in your infra

**Reply by eglaost** [reply]
We are using AWS. We created a volume that is mounted on each pod and includes the build directory. Each next.js instance uses this directory but still... it's not working. Although after the first pod is triggered to generate new static files they are stored in the directory, (modified dates are changed), when the second pod is called it still serves the old content. It's like next doesn't serve files from the directory even though it use it to store them

**Comment by akellbl4** [comment]
I had the same issue. In my case, Next.js was not able to write on disk.
Try to check if the user that runs the next.js instance has enough privileges to write files on disk

**Comment by imadatyatalah** [comment]
I have solved this problem using [react-query with hydration](https://react-query.tanstack.com/guides/ssr#using-hydration)
